dependencies.yaml

```diff
@@ -0,0 +1,124 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
+  all:
+    output: conda
+    matrix:
+      cuda: ["11.5"]
+      arch: [x86_64]
+    includes:
+      - build
+      - cudatoolkit
+      - develop
+      - doc
+      - notebook
+      - run
+      - test_python
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - dask/label/dev
+  - conda-forge
+  - nvidia
+dependencies:
+  build:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - cmake>=3.23.1,!=3.25.0
+          - cuda-python >=11.7.1,<12.0
+          - cython>=0.29,<0.30
+          - ninja
+          - scikit-build>=0.13.1
+      - output_types: [conda]
+        packages:
+          - c-compiler
+          - cxx-compiler
+          - rapids-build-env=23.02.*
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - gcc_linux-64=9.*
+              - sysroot_linux-64==2.17
+          - matrix:
+              arch: aarch64
+            packages:
+              - gcc_linux-aarch64=9.*
+              - sysroot_linux-aarch64==2.17
+  develop:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - clang=11.1.0
+      - output_types: [conda]
+        packages:
+          - clang-tools=11.1.0
+  cudatoolkit:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+  doc:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - breathe
+      - output_types: [requirements]
+        packages:
+          - sphinx_markdown_tables
+      - output_types: [conda]
+        packages:
+          - doxygen>=1.8.20
+          - rapids-doc-env=23.02.*
+          - sphinx-markdown-tables
+  notebook:
+    common:
+      - output_types: [conda]
+        packages:
+          - rapids-notebook-env=23.02.*
+  py_version:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              py: "3.8"
+            packages:
+              - python=3.8
+          - matrix:
+              py: "3.9"
+            packages:
+              - python=3.9
+          - matrix:
+            packages:
+              - python>=3.8,<3.10
+  run:
+    common:
+      - output_types: [conda]
+        packages:
+          - rmm=23.02.*
+          - dask==2022.11.1
+          - distributed==2022.11.1
+          - ucx>=1.13.0
+          - ucx-py=0.30.*
+          - ucx-proc=*=gpu
+          - libfaiss>=1.7.0
```

Comment:
I noticed a really annoying issue recently in rapids-compose. It merges the dependency lists / conda channels, and ends up placing the `pytorch` channel (which is used in cudf tests) above `conda-forge`. This in turn means that `libfaiss` is pulled from `pytorch` and it ignores the `faiss-proc=*=cuda` package below, which only affects `libfaiss` from `conda-forge`. 🙃 Just something to be aware of. We may need to specify `libfaiss>=1.7.0=cuda*` or something like that, in addition to `faiss-proc=*=cuda`. Another solution would be to fix the conda-forge packages of `pytorch` and never use the `pytorch` channel for any dependencies.

Reply:
I just ran into the same issue and fixed it in the old env files, but then I realized this PR was ready to go. I'll make a PR with the additional pinning, which is how I fixed it in my case and it worked fine.

Reply:
#1078